From 2c004384d0a7c6e4b65b7a4e4e913a86750a1568 Mon Sep 17 00:00:00 2001
From: Nagashree S <nagashre@codeaurora.org>
Date: Thu, 9 Nov 2017 17:38:11 +0530
Subject: [PATCH 2/2] BT: Send intent for the call type when call is active.

The change checks the call capabilities to find out if its of
type CS and sends intent for receiver in bluetooth application.

CRs-Fixed: 977030
Change-Id: Ie0d8aea7083be712ddb5b176df444fd87fcf6705
---
 .../android/server/telecom/BluetoothPhoneServiceImpl.java    | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/com/android/server/telecom/BluetoothPhoneServiceImpl.java b/src/com/android/server/telecom/BluetoothPhoneServiceImpl.java
index 0046bec..eea05f7 100644
--- a/src/com/android/server/telecom/BluetoothPhoneServiceImpl.java
+++ b/src/com/android/server/telecom/BluetoothPhoneServiceImpl.java
@@ -35,6 +35,9 @@ import android.telecom.VideoProfile;
 import android.telephony.PhoneNumberUtils;
 import android.telephony.TelephonyManager;
 import android.text.TextUtils;
+import android.os.UserHandle;
+
+import android.telecom.TelecomManager;
 
 import com.android.internal.annotations.VisibleForTesting;
 import com.android.server.telecom.CallsManager.CallsManagerListener;
@@ -776,6 +779,15 @@ public class BluetoothPhoneServiceImpl {
             boolean sendDialingFirst = mBluetoothCallState != bluetoothCallState &&
                     bluetoothCallState == CALL_STATE_ALERTING;
 
+            if (numActiveCalls > 0) {
+                    Log.i(TAG, "updateHeadsetWithCallState: Call active");
+                    boolean isCsCall = ((activeCall != null) &&
+                            !(activeCall.can(Connection.PROPERTY_HIGH_DEF_AUDIO) ||
+                            activeCall.can(Connection.PROPERTY_WIFI)));
+                    final Intent intent = new Intent(TelecomManager.ACTION_CALL_TYPE);
+                    intent.putExtra(TelecomManager.EXTRA_CALL_TYPE_CS, isCsCall);
+                    mContext.sendBroadcastAsUser(intent, UserHandle.ALL);
+            }
             mOldHeldCall = heldCall;
             mNumActiveCalls = numActiveCalls;
             mNumChildrenOfActiveCall = numChildrenOfActiveCall;
-- 
2.7.4

